builds/linux/debian/11/linux-debian.pkr.hcl

```diff
@@ -0,0 +1,180 @@
+*/
+
+//  BLOCK: packer
+//  The Packer configuration.
+
+packer {
+  required_version = ">= 1.8.0"
+  required_plugins {
+    vsphere = {
+      version = ">= v1.0.3"
+      source  = "github.com/hashicorp/vsphere"
+    }
+  }
+}
+
+//  BLOCK: locals
+//  Defines the local variables.
+
+locals {
+  build_by      = "Built by: HashiCorp Packer ${packer.version}"
+  build_date    = formatdate("YYYY-MM-DD hh:mm ZZZ", timestamp())
+  build_version = formatdate("YY.MM", timestamp())
+  manifest_date = formatdate("YYYY-MM-DD hh:mm:ss", timestamp())
+  manifest_path = "${path.cwd}/manifests/"
+  data_source_content = {
+    "/ks.cfg" = templatefile("${abspath(path.root)}/data/ks.pkrtpl.hcl", {
+      build_username           = var.build_username
+      build_password_encrypted = var.build_password_encrypted
+      vm_guest_os_language     = var.vm_guest_os_language
+      vm_guest_os_keyboard     = var.vm_guest_os_keyboard
+      vm_guest_os_timezone     = var.vm_guest_os_timezone
+    })
+  }
+}
+
+//  BLOCK: source
+//  Defines the builder configuration blocks.
+
+source "vsphere-iso" "linux-debian" {
+
+  // vCenter Server Endpoint Settings and Credentials
+  vcenter_server      = var.vsphere_endpoint
+  username            = var.vsphere_username
+  password            = var.vsphere_password
+  insecure_connection = var.vsphere_insecure_connection
+
+  // vSphere Settings
+  datacenter = var.vsphere_datacenter
+  cluster    = var.vsphere_cluster
+  datastore  = var.vsphere_datastore
+  folder     = var.vsphere_folder
+
+  // Virtual Machine Settings
+  guest_os_type        = var.vm_guest_os_type
+  vm_name              = "${var.vm_guest_os_family}-${var.vm_guest_os_name}-${var.vm_guest_os_version}-v${local.build_version}"
+  firmware             = var.vm_firmware
+  CPUs                 = var.vm_cpu_sockets
+  cpu_cores            = var.vm_cpu_cores
+  CPU_hot_plug         = var.vm_cpu_hot_add
+  RAM                  = var.vm_mem_size
+  RAM_hot_plug         = var.vm_mem_hot_add
+  cdrom_type           = var.vm_cdrom_type
+  disk_controller_type = var.vm_disk_controller_type
+  storage {
+    disk_size             = var.vm_disk_size
+    disk_thin_provisioned = var.vm_disk_thin_provisioned
+  }
+  network_adapters {
+    network      = var.vsphere_network
+    network_card = var.vm_network_card
+  }
+  vm_version           = var.common_vm_version
+  remove_cdrom         = var.common_remove_cdrom
+  tools_upgrade_policy = var.common_tools_upgrade_policy
+  notes                = "Version: v${local.build_version}\nBuilt on: ${local.build_date}\n${local.build_by}"
+
+  // Removable Media Settings
+  iso_paths    = ["[${var.common_iso_datastore}] ${var.iso_path}/${var.iso_file}"]
+  iso_checksum = "${var.iso_checksum_type}:${var.iso_checksum_value}"
+  http_content = var.common_data_source == "http" ? local.data_source_content : null
+  cd_content   = var.common_data_source == "disk" ? local.data_source_content : null
```

Comment:
Have tests been run using both `http_content` and `cd_content` methods by changing `http` to `disk` in the common vars?

Reply:
No actually, i left this as it was.

Can you explain me in more details the effect of this variables and the configurations that you would like me to test?

Reply:
I notice that the `data_source_command` is missing in the `locals` and that it's not included in the manifest.

Example:
```hcl
locals {
  #....
  data_source_command = var.common_data_source == "http" ? "url=http://{{ .HTTPIP }}:{{ .HTTPPort }}/ks.cfg" : "file=cdrom:/ks.cfg"
}
```

So that it can be called based on setting `var.common_data_source` to `http` (default) or `disk`.

Example:

```hcl
  boot_command = [
    "<wait3s>c<wait3s>",
    "linux /install.amd/vmlinuz",
    " auto-install/enable=true",
    " debconf/priority=critical",
    " ${local.data_source_command}",
    " noprompt --<enter>",
    "initrd /install.amd/initrd.gz<enter>",
    "boot<enter>"
  ]
```

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.

Reply:
I see, would you like to patch this yourselves?

Reply:
p.s. i just noticed that all the Ubuntu packer templates misses to use "var.build_password"; Due to this reason the only way to set a password is currently to define it in a configuration file but if the users use the set-var.sh script the variable wont be used.

To replicate it just run a build removing the build_password from the common_configuration file. You will notice that doing that way packer will miss the variable to authenticate even if this will be set with the set-var.sh script.

Reply:
I've just now tested the above example whilst I had a moment and the results were as seen below:

<img width="933" alt="image" src="https://user-images.githubusercontent.com/7771363/163212253-22f35c11-10f4-485c-bdca-cbc551c49da8.png">

It seems to find the `ks.cfg` but is hitting an issue. 

Thd project aims to ensure both the `http` and `disk` sources can be used where there may be environments that can't route back to the Packer HTTP server. If you could take a look into it more, I'd appreciate it. Meanwhile, I was looking at the inclusion of the Ansible portions when time is available. I had most of that working last evening after making a few minor edits to your branch.

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.

Reply:
> p.s. i just noticed that all the Ubuntu packer templates misses to use "var.build_password"; Due to this reason the only way to set a password is currently to define it in a configuration file but if the users use the set-var.sh script the variable wont be used.
> 
> To replicate it just run a build removing the build_password from the common_configuration file. You will notice that doing that way packer will miss the variable to authenticate even if this will be set with the set-var.sh script.

Please reference this in a new issue (`bug`). 🐞 

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.